File: Karthik/14-9_to_15-9/mostwater.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        i=0
        j=len(height)-1
        arraymax=0
        while i<j:
            minimum=min(height[i],height[j])
            if minimum*(j-i)>arraymax:
                arraymax=minimum*(j-i)
            if minimum==height[i]:
                i+=1
            else:
                j-=1
        return arraymax
        
            
        

File: Nithya/14-09-22/mostwater.cpp
class Solution {
public:
    int maxArea(vector<int>& height)
    {
        int i=0;
        int j=height.size()-1;
        int max_area=0,m1;
        while(i<j)
        {
            m1=min(height[i],height[j])*(j-i);
            max_area=max(max_area,m1);
            height[i]<height[j]?i++:j--;
        }
        return max_area;
    }
};